resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -249,6 +259,10 @@ private[spark] class Client(
     val capability = Records.newRecord(classOf[Resource])
     capability.setMemory(amMemory + amMemoryOverhead)
     capability.setVirtualCores(amCores)
+    if (amResources.nonEmpty) {
```

Comment:
Nitpick: I would execute `ResourceTypeHelper.setResourceInfoFromResourceTypes(amResources, capability)` unconditionally. That's just my style, though.

Reply:
So the goal of my original comments was to avoid noisy logs. There were too many before, and there is none now. Neither is a good place to be.

You want to log a warning message if the user is setting these properties but the YARN libraries don't support the feature.